### PR TITLE
Implemented things by branch name, also:

### PR DESCRIPTION
**Pull request by b0r1ngx**
1) Created setContentDescription modifier that allows to set contentDescription on any used Modifier - as accessibility feature, suppossing that it may help to understand the UI architecture components on the screen more

Not tested how it would work for mines >999, or <-99, things that code logic is draw more digits, how much it needs

### CODE REVIEW

**Comment by b0r1ngx** [comment]
To understand how it looks like, attach few screenshots:
When start a game:
image

when user mark more than max of the mines:
image

**Comment by b0r1ngx** [comment]
@arkivanov Hello, please, check how it looks like!

File: composeApp/src/commonMain/kotlin/com/arkivanov/minesweeper/game/GameState.kt [hunk 1]
```diff
@@ -98,5 +98,6 @@ internal val CellStatus.isFlagged: Boolean
 internal fun Cell.open(): Cell =
     copy(status = CellStatus.Open)
 
+// TODO: Can we somehow increase work of this thing? not each time count run for all grid
```
Comment (arkivanov) [inline composeApp/src/commonMain/kotlin/com/arkivanov/minesweeper/game/GameState.kt]: Let's change the TODO text to something like "Consider making this a part of the state"

File: composeApp/src/commonMain/kotlin/com/arkivanov/minesweeper/ComposeUtils.kt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package com.arkivanov.minesweeper
+
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.semantics.role
+import androidx.compose.ui.semantics.semantics
+
+fun Modifier.setContentDescription(description: String, role: Role): Modifier =
```
Comment (arkivanov) [inline composeApp/src/commonMain/kotlin/com/arkivanov/minesweeper/ComposeUtils.kt]: Let's name it as `semantics`.

File: composeApp/src/commonMain/kotlin/com/arkivanov/minesweeper/game/GameContent.kt [hunk 3]
```diff
@@ -126,6 +131,43 @@ internal fun GameContent(component: GameComponent, modifier: Modifier = Modifier
     }
 }
 
+@Composable
+fun RemainingMines(
```
Comment (arkivanov) [inline composeApp/src/commonMain/kotlin/com/arkivanov/minesweeper/game/GameContent.kt]: I think we can leverage the `padStart` function and do something as follows. This `Counter` function could be reused in both cases - remaining mine counter and (future) time counter.

```kotlin
@Composable
private fun Counter(value: Int, modifier: Modifier = Modifier) {
    Row(modifier = modifier, horizontalArrangement = Arrangement.Center) {
        value.toCounterString().forEach { char ->
            Image(
                modifier = Modifier.size(width = 13.dp, height = 23.dp),
                painter = LocalGameIcons.icons.digits.getValue(char),
                contentDescription = char.toString(),
            )
        }
    }
}

private fun Int.toCounterString(): String =
    if (this >= 0) {
        coerceAtMost(999).toString().padStart(length = COUNTER_LENGTH, padChar = '0')
    } else {
        "-" + coerceAtLeast(-99).absoluteValue.toString().padStart(length = COUNTER_LENGTH - 1, padChar = '0')
    }
```

File: composeApp/src/commonMain/kotlin/com/arkivanov/minesweeper/game/GameContent.kt [hunk 4]
```diff
@@ -126,6 +131,43 @@ internal fun GameContent(component: GameComponent, modifier: Modifier = Modifier
...
+        horizontalArrangement = Arrangement.Center
+    ) {
+        val zeroesToDrawOnTheRightSide = defaultCountersSize - digits
+        if (zeroesToDrawOnTheRightSide > 0) {
+            for (i in 0 until zeroesToDrawOnTheRightSide) {
+                Image(
+                    painter = LocalGameIcons.icons.digits.getValue('0'),
+                    contentDescription = "0",
+                    contentScale = ContentScale.Crop,
+                    modifier = Modifier.height(26.dp),
```
Comment (arkivanov) [inline composeApp/src/commonMain/kotlin/com/arkivanov/minesweeper/game/GameContent.kt]: Let's define the exact size for the image as per PNGs: 13.dp x 23.dp

File: composeApp/src/commonMain/kotlin/com/arkivanov/minesweeper/game/GameContent.kt [hunk 5]
```diff
@@ -52,8 +52,11 @@ import androidx.compose.ui.unit.dp
 import com.arkivanov.decompose.extensions.compose.subscribeAsState
 import com.arkivanov.decompose.value.MutableValue
 import com.arkivanov.decompose.value.Value
+import com.arkivanov.minesweeper.countValuebleChars
+import com.arkivanov.minesweeper.setContentDescription
 
 private val cellSize = 16.dp
+private const val defaultCountersSize = 3
```
Comment (arkivanov) [inline composeApp/src/commonMain/kotlin/com/arkivanov/minesweeper/game/GameContent.kt]: I suggest to name it as `COUNTER_LENGTH`.

File: composeApp/src/commonMain/kotlin/com/arkivanov/minesweeper/ComposeUtils.kt [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    this.then(
```
Comment (arkivanov) [inline composeApp/src/commonMain/kotlin/com/arkivanov/minesweeper/ComposeUtils.kt]: Can we just call `semantics { ... }` here?

**Review by arkivanov — CHANGES_REQUESTED** [review]

**Comment by b0r1ngx** [comment]
@arkivanov Hello, I'm made recommended changes, please, check how it looks like!

File: composeApp/src/commonMain/kotlin/com/arkivanov/minesweeper/game/GameContent.kt [hunk 7]
```diff
@@ -72,11 +74,12 @@ internal fun GameContent(component: GameComponent, modifier: Modifier = Modifier
...
                     horizontalArrangement = Arrangement.SpaceAround
                 ) {
-                    Text(
-                        text = "$remainingBombs",
-                        style = MaterialTheme.typography.body2,
-                        modifier = Modifier.weight(1f),
-                        textAlign = TextAlign.Center
+                    Counter(
+                        value = remainingMines,
+                        modifier = Modifier.weight(1f).setSemantics(
```
Comment (arkivanov) [inline composeApp/src/commonMain/kotlin/com/arkivanov/minesweeper/game/GameContent.kt]: Now after the changes, maybe we can just use the original `semantics` modifier? What benefits the custom function gives us?

File: composeApp/src/commonMain/kotlin/com/arkivanov/minesweeper/game/GameContent.kt [hunk 8]
```diff
@@ -86,12 +89,7 @@ internal fun GameContent(component: GameComponent, modifier: Modifier = Modifier
                         onClick = component::onRestartClicked,
                     )
 
-                    Text(
-                        text = "000", // TODO: Place for stopwatch
-                        style = MaterialTheme.typography.body2,
-                        modifier = Modifier.weight(1f),
-                        textAlign = TextAlign.Center
-                    )
+                    Stopwatch(modifier = Modifier.weight(1f))
```
Comment (arkivanov) [inline composeApp/src/commonMain/kotlin/com/arkivanov/minesweeper/game/GameContent.kt]: Let's just use `Counter(value = 0, modifier = Modfier.weight(1F))` here for now.

**Review by arkivanov — CHANGES_REQUESTED** [review]

**Comment by b0r1ngx** [comment]
Let's rock it!

**Review by arkivanov — APPROVED** [review]

**Comment by arkivanov** [comment]
Thank you!

**Comment by b0r1ngx** [comment]
> Thank you!

Thank you too, for reviewing my code and making it better together! 

Hope I will start implementing the stopwatch soon, I don't want to use additional dependencies (for educational purposes for me and project, if someone wants to research it)

Also I tried to go to wasm site from this repo, I also got amazed that you create a pipeline that automatically creates a build and publishes it to the site (you did a great work with this CI/CD)

I visit it from mobile android chrome browser, and can't click anything. is this wasm restriction for now? Or it just doesn't work because in the project clicks are used instead of touches?

**Comment by arkivanov** [comment]
Yeah, Wasm on Android chrome is pretty much broken currently. Also on any browser, all mouse clicks are recognised as left clicks.

**Pull request merged**